R/superPartition.R

```diff
@@ -0,0 +1,268 @@
+#' prt
+#'
+#' @references
+#' Barrett, Malcolm and Joshua Millstein (2020). partition: A fast and flexible framework for data reduction in R. Journal of Open Source Software, 5(47), 1991, https://doi.org/10.21105/joss.01991Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. *Bioinformatics* **36** (2019) 676–681. doi:10.1093/bioinformatics/btz661.
+#'
+#' Gagolewski, Marek, Maciej Bartoszuk, and Anna Cena. "Genie: A new, fast, and outlier-resistant hierarchical clustering algorithm." Information Sciences 363 (2016): 8-23.
+#'
+#' Millstein, Joshua, Francesca Battaglin, Malcolm Barrett, Shu Cao, Wu Zhang, Sebastian Stintzing, Volker Heinemann, and Heinz-Josef Lenz. 2020. “Partition: A Surjective Mapping Approach for Dimensionality Reduction.” *Bioinformatics* 36 (3): https://doi.org/676–81.10.1093/bioinformatics/btz661.
+#'
+#' @seealso [partition()]
+#'
+#' @author Katelyn Queen, \email{kjqueen@@usc.edu}
+#'
+#' @export
+#' @import progress
+#' @import genieclust
+#' @import gtools
+#' @import tibble
+super_partition <- function(full_data,
+                            threshold = 0.5,
+                            cluster_size = 4000,
+                            partitioner = part_icc(),
+                            tolerance = .0001,
+                            niter = NULL,
+                            x = "reduced_var",
+                            .sep = "_") {
+
+  # ensure 0 < threshold < 1
+  if(0 > threshold | 1 < threshold) stop("Threshold must be between 0 and 1.")
+
+  # ensure data frame structure
+  full_data <- as.data.frame(full_data)
+
+  # if < 4,000 features, call regular partition
+  if(ncol(full_data) < cluster_size) {
+    message(paste0("Using regular Partition function since there are < ", cluster_size, "features."))
+    return(partition(full_data, threshold = threshold))
+  }
+
+  # iteration counters
+  i = j = k = l = 0
+
+  # Function to go from cluster-specific column numbers to full data column numbers
+  ## full_data  - sample by probe expression data for all probes
+  ## small_data - sample by probe expression data for probes in specific cluster
+  ## modules   - indices from partition mapping key
+  ## return - indices from partition mapping key for full data
+  full_data_col_numbers <- function(full_data, small_data, modules) {
+
+    # vector of lists to return
+    return_mods <- numeric(length(modules))
+
+    for (i in 1:length(modules)) {
+      # create vector for full data indices
+      col_full <- numeric(length(unlist(modules[i])))
+
+      # get column names
+      col_small <- colnames(small_data)[unlist(modules[i])]
+
+      # get column indices in full data
+      for (j in 1:length(col_small)) {
+        col_full[j] <- grep(col_small[j], colnames(full_data))
+      }
+
+      # add list of indices to return vector
+      return_mods[i] <- list(col_full)
+    }
+
+    return_mods <- return_mods
```

Comment:
This doesn't appear to be necessary. Do you mean to assign another object to `return_mods`?

Reply:
You had asked to remove explicit return() statements, which is what used to be here. If I remove this line, then nothing returns from the function.

Reply:
That doesn't seem right. `return_mods` alone should return `return_mods`. Can you investigate a bit more?

Reply:
I don't know why it didn't occur to me to just put return_mods... this is updated now and works correctly!